### PR TITLE
roachtest: update multitenant-multiregion to use roachprod API

### LINKED ISSUE #124029: roachtest: port multitenant-multiregion to new APIs
See the parent issue [1] for details.
Test location: `pkg/cmd/roachtest/tests/multitenant.go`

[1] https://github.com/cockroachdb/cockroach/issues/115867

Epic: [CRDB-31933](https://cockroachlabs.atlassian.net/browse/CRDB-31933)

Jira issue: CRDB-35299

Jira issue: CRDB-38669

### PR DESCRIPTION
**Pull request by renatolabs**
This updates the `multitenant-multiregion` test to use the "official"
multitenant roachprod API instead of the deprecated functions in
`multitenant_utils.go`.

Fixes: #124029

Release note: None

### CODE REVIEW

**Comment by renatolabs** [comment]
Test passed 5 times on this branch: [link](https://teamcity.cockroachdb.com/viewLog.html?buildId=16809595&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=buildResultsDiv&branch_Cockroach_Nightlies=130178).

It would be helpful to get a review from @cockroachdb/sql-foundations as well. I made a minor update to use a "real" workload (YCSB in this case) instead of the previous `CREATE TABLE foo`. The latter makes more sense in unit tests, not end-to-end tests such as this one.

For context, once this is merged, `multitenant-upgrade` will be the last test to use the deprecated multitenant functions, and that test will be phased out once the `mixedversion` framework is able to generate separate-process upgrades (#130194).

**Review by herkolategan — APPROVED** [review]

**Review by fqazi — CHANGES_REQUESTED** [review]
Overall this is a lot cleaner, I did have one question

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130178#-:-O6g4kNUD4CCS6PkEXZn:bttej80)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/multitenant.go` line 359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130178#-O6g4Rh08ZVEfZTPsyQP:-O6g4Rh1AYeUn6tDmk1y:b-ikvoan) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bfa2840904b62594f61cb52bcd83349b792035c/pkg/cmd/roachtest/tests/multitenant.go#L359)):*
> ```Go
> 	killedRegion := c.Range(otherRegionNode, len(c.All()))
> 	t.Status("stopping the server ahead of checking for the tenant server")
> 	c.Stop(ctx, t.L(), option.DefaultStopOpts(), killedRegion)
> ```

Will this cause a graceful stop? Might be good to confirm things are draining, since we were trying to simulate the loss of a region.

**Review by renatolabs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130178)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/multitenant.go` line 359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130178#-O6g4Rh08ZVEfZTPsyQP:-O6gcNjyAbDFDTA56Dej:b5d7wmw) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bfa2840904b62594f61cb52bcd83349b792035c/pkg/cmd/roachtest/tests/multitenant.go#L359)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Will this cause a graceful stop? Might be good to confirm things are draining, since we were trying to simulate the loss of a region.
</blockquote></details>

It's not a gracefull stop, mostly because I was preserving the behaviour of the existing test, which does a `kill -9`:

https://github.com/cockroachdb/cockroach/blob/259cde8ae0bdde90f2bb491357f0caef99a4dfc2/pkg/cmd/roachtest/tests/multitenant.go#L335

Do you want me to change this to do graceful stops?

**Review by fqazi — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/130178#-:-O6gclXl61xXEJbpvBQJ:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)

___
*[`pkg/cmd/roachtest/tests/multitenant.go` line 359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/130178#-O6g4Rh08ZVEfZTPsyQP:-O6gchX87lwRVO9LhF1t:b-t7altv) ([raw file](https://github.com/cockroachdb/cockroach/blob/0bfa2840904b62594f61cb52bcd83349b792035c/pkg/cmd/roachtest/tests/multitenant.go#L359)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

It's not a gracefull stop, mostly because I was preserving the behaviour of the existing test, which does a `kill -9`:

https://github.com/cockroachdb/cockroach/blob/259cde8ae0bdde90f2bb491357f0caef99a4dfc2/pkg/cmd/roachtest/tests/multitenant.go#L335

Do you want me to change this to do graceful stops?
</blockquote></details>

Nope, we want non-graceful. That's perfect!

**Pull request merged**
